sway/meson.build

```diff
@@ -58,6 +58,8 @@ sway_sources = files(
 	'commands/swaybg_command.c',
 	'commands/workspace.c',
 	'commands/ws_auto_back_and_forth.c',
+	'commands/lock.c',
+	'commands/swaylock_command.c',
```

Comment:
These should be taken out, and swayidle shouldn't use sway IPC directly. It should just let the user execute commands on a timeout, such as `swaymsg exec swaylock ...`

Reply:
Well I understand that... but I really think that it would be nice for the user to have a consistent way to run the lockscreen, so it looks the same everytime. And for the systemd handling that executes the lockscreen, it is impossible to know how the user wants it. So it makes sense have a sway command for that, so that it is possible to just let sway lock the screen and get the same result everytime. This also have the benefit that the user can bind a key to the lock command and get the same lockscreen as swayidle uses for the systemd part.

Reply:
>I really think that it would be nice for the user to have a consistent way to run the lockscreen, so it looks the same everytime

Just run it with the same arguments every time, then. If you want to run it from several places with the same arguments without having to update all of them every time you change it, wrap swaylock in a script.

Reply:
Well, since the sleep lock part, is nothing we get from the user (since that will make it impossible to know how to verify it is ok to release the lock), it is not possible to get this from the user. This is the reason I added the lock and swaylock_command.

Reply:
What is the "sleep lock" part?

Reply:
Is there a problem having a lock and swaylock_command? Or is there a problem with swayidle doing ipc? I'm not really sure I understand the issues caused by this?

Reply:
There is, yes. I intend to split swaylock out into a standalone project that works on many compositors. Users might want to lock with a different lock screen as well. It also adds unnecessary complexity, it's no better than just having swayidle take an arbitrary command.

Reply:
I assume that if you can help me make the IPC_IS_LOCKSCREEN_ACTIVE more generic (currently it looks at the name of the lockscreen surface), I guess we could just let the -x parameter take the lock command to use.

Reply:
I don't think IPC has to be involved here at all. A parameter specifying the lockscreen would be good but not sure why swayidle has to know the command it's running is a lockscreen at all